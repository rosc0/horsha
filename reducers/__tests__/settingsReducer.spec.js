import reducer, { initialState as state } from '../settings'
import types from '../../constants/actionTypes'

const expectResponses = {
  status: {
    language: 'en',
    pushNotificationsEnabled: false,
    statusBarHidden: undefined,
  },
  push: {
    language: 'en',
    pushNotificationsEnabled: true,
    statusBarHidden: false,
  },
  language: {
    language: 'pt-br',
    pushNotificationsEnabled: false,
    statusBarHidden: false,
  },
}

describe('Settings Reducer', () => {
  it('should return initial state', () => {
    expect(reducer()).toBe(state)
    expect(reducer()).toMatchSnapshot()
  })

  it('should CHANGE_LANGUAGE', () => {
    const act = reducer(state, {
      type: types.CHANGE_LANGUAGE,
      payload: 'pt-br',
    })

    expect(act).toEqual(expectResponses.language)
    expect(act).toMatchSnapshot()
  })

  it('should TOGGLE_STATUS_BAR', () => {
    const act = reducer(state, {
      type: types.TOGGLE_STATUS_BAR,
    })

    expect(act).toEqual(expectResponses.status)
    expect(act).toMatchSnapshot()
  })

  it('should SETTINGS_ENABLE_PUSH_NOTIFICATIONS', () => {
    const act = reducer(state, {
      type: types.SETTINGS_ENABLE_PUSH_NOTIFICATIONS,
    })

    expect(act).toEqual(expectResponses.push)
    expect(act).toMatchSnapshot()
  })
})
