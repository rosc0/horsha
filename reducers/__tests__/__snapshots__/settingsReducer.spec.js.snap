// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings Reducer should CHANGE_LANGUAGE 1`] = `
Object {
  "language": "pt-br",
  "pushNotificationsEnabled": false,
  "statusBarHidden": false,
}
`;

exports[`Settings Reducer should SETTINGS_ENABLE_PUSH_NOTIFICATIONS 1`] = `
Object {
  "language": "en",
  "pushNotificationsEnabled": true,
  "statusBarHidden": false,
}
`;

exports[`Settings Reducer should TOGGLE_STATUS_BAR 1`] = `
Object {
  "language": "en",
  "pushNotificationsEnabled": false,
  "statusBarHidden": undefined,
}
`;

exports[`Settings Reducer should return initial state 1`] = `
Object {
  "language": "en",
  "pushNotificationsEnabled": false,
  "statusBarHidden": false,
}
`;
